src/components/Listing.tsx

```diff
@@ -1,37 +1,52 @@
 import React from "react";
 import { ListItem } from "../types/ListItem";
 import { ListingItem } from "../components/ListingItem";
+import { Tag } from "../types/Tag";
 
 export type ListingProps = {
   items: ListItem[];
+  knownTags: Tag[];
   title: "Previously read links" | "Links to read";
   itemChanged: (listItem: ListItem) => void;
+  tagAdded: (item: ListItem, tag: Tag) => void;
+  tagDeleted: (item: ListItem, tag: Tag) => void;
+  updateKnownTags: (title: string) => Tag | undefined;
+};
+export const Listing: (props: ListingProps) => JSX.Element = ({
+  items,
+  knownTags,
+  title,
+  itemChanged,
+  tagAdded,
+  tagDeleted,
+  updateKnownTags,
+}: ListingProps) => {
```

Comment:
Could be simplified even more if you want to.
E.g.
`const foo = () => { return (<text>Bar</text>) }` equals to `const foo = () => (<text>Bar</text>)`. See implicit return part of [this article](https://medium.com/@bunlong/arrow-functions-return-rules-in-javascript-b63ed5f25994).

Reply:
Great!